pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -88,12 +87,11 @@ func (pl *NodeAffinity) Score(ctx context.Context, state *framework.CycleState,
 			}
 
 			// TODO: Avoid computing it for all nodes if this becomes a performance problem.
-			nodeSelector, err := v1helper.NodeSelectorRequirementsAsSelector(preferredSchedulingTerm.Preference.MatchExpressions)
+			matches, err := corev1.MatchNodeSelectorTerms(node, []v1.NodeSelectorTerm{preferredSchedulingTerm.Preference})
```

Comment:
here's an example of where returning `true, error` would be a behavior change we probably wouldn't want

Reply:
In this case I think the behavior is the same. We were previously returning an error if the selector failed to parse before we made the call to check if it matches, this just combines it into one step.

(Also, at least in the scheduler side we want to fail early with invalid selectors since they lack API validation, see https://github.com/kubernetes/kubernetes/pull/93660)

Reply:
> We were previously returning an error if the selector failed to parse before we made the call to check if it matches

ah, true